### PR TITLE
kubeletcontroller: minimize and tune reconciliation loop

### PR DESCRIPTION
**Pull request by Tal-or**
- KubeletConfig deletion event:
  **old behavior:** infinite loops trying to fetch the already deleted object.
  **new behavior:** do nothing but logging out the fact that the resource got deleted.

- ConfigMap update/deletion event:
  **old behavior:** operator didn't detect external/manual tamper and keep the resource in undesired state.
  **new behavior:** set KubeletConfig as the OwnerReference which triggers the correct reconciliation loop and keep the resource (ConfigMap) in the desired state

### CODE REVIEW

**Review by fromanirh — COMMENTED** [review]
/lgtm
/approve
very nice!

**Review by fromanirh — COMMENTED** [review]
is this still WIP? because it LGTM :)

File: test/e2e/rte/rte_test.go [hunk 1]
```diff
@@ -196,6 +197,61 @@ var _ = ginkgo.Describe("with a running cluster with all the components", func()
...
+			klog.Infof("generated config map name: %q", generatedName)
+			cm, err := f.ClientSet.CoreV1().ConfigMaps(namespace).Get(context.TODO(), generatedName, metav1.GetOptions{})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+			desiredMapState := make(map[string]string)
+			for k, v := range cm.Data {
+				desiredMapState[k] = v
+			}
+
+			cm.Data = nil
```
[the remark this answers is not in the document]
Comment (Tal-or) [reply test/e2e/rte/rte_test.go:240]: I didn't want start hardcoded keys so.. :)

[the remark this answers is not in the document]
Comment (fromanirh) [reply test/e2e/rte/rte_test.go:240]: that's fair enough

**Review by fromanirh — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**